components/places/src/api/places_api.rs

```diff
@@ -395,16 +376,31 @@ impl PlacesApi {
         let _guard = self.sync_state.lock().unwrap();
         let conn = self.open_sync_connection()?;
 
-        // Somewhat ironically, we start by migrating from the legacy storage
-        // format. We *are* just going to delete it anyway, but the code is
-        // simpler if we can just reuse the existing path.
-        HistoryEngine::migrate_v1_global_state(&conn)?;
-
         history_sync::reset(&conn, &sync15::EngineSyncAssociation::Disconnected)?;
         Ok(())
     }
 
-    /// Get a new interrupt handle for the sync connection.
+    /// Create a new SqlInterruptScope for syncing
+    ///
+    /// Call this at the begining of a sync operation.  Then if interrupt_sync() is called, we will
+    /// interrupt the current DB query and return `InterruptedError`
+    pub fn begin_sync_interrupt_scope(&self) -> SqlInterruptScope {
+        SqlInterruptScope::new(self.sync_interrupt_counter.clone())
+    }
+
+    /// Interrupt the current sync operation if one is in progress
+    pub fn interrupt_sync(&self) {
+        // TODO: update the sync connection code so we can implement this method
```

Comment:
This all looks great - except this bit :) I'll have a deeper look to reacquaint myself with that interrupt stuff soon, but this is certainly a nice improvement.